pkg/reconciler/route/resources/ingress.go

```diff
@@ -127,6 +130,7 @@ func MakeIngressSpec(
 
 	// The routes are matching rule based on domain name to traffic split targets.
 	rules := make([]v1alpha1.IngressRule, 0, len(names))
```

Comment:
Preallocate enough space for both `names` and `additionalRules` here?

`rules := make([]v1alpha1.IngressRule, 0, len(names) + len(additionalRules))`